test/e2e/workflow/deployment_test.go

```diff
@@ -151,25 +154,31 @@ var _ = Describe("NetworkAddonsConfig", func() {
 		BeforeEach(func() {
 			CreateConfig(configSpec)
 			CheckConfigCondition(ConditionAvailable, ConditionTrue, 15*time.Minute, CheckDoNotRepeat)
+			CheckAvailableEvent()
 		})
 		//2305
 		It("should remain in Available condition after applying the same config", func() {
 			UpdateConfig(configSpec)
+			CheckModifiedEvent()
+			CheckProgressingEvent()
 			CheckConfigCondition(ConditionAvailable, ConditionTrue, CheckImmediately, 30*time.Second)
 		})
 		//2281
 		It("should be able to remove all of them by removing the config", func() {
 			DeleteConfig()
 			CheckComponentsRemoval(components)
+			CheckModifiedEvent()
```

Comment:
We don't need to test it in every single test case. In general, you want to have one test for each feature. When it is done like that, it is easier to identify which feature is broken.

Do we have a suiting tests for status changes? You should make sure that we don't test events multiple times in identical scenarios

Alternatively

Reply:
> We don't need to test it in every single test case. In general, you want to have one test for each feature. When it is done like that, it is easier to identify which feature is broken.
> 

I understand. So I think the recovery test covers all types of event creation. I'll keep only this now
> Do we have a suiting tests for status changes?

I think this is tested along with the integration tests. I don't think there is a  dedicated suit if that's what you have meant.